src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -28,15 +28,16 @@ public static bool IsProcessRunning(int processId)
         public static bool IsProcessRunning(int processId, string machineName)
         {
             // Performance optimization for the local machine:
-            // First try to OpenProcess by id, if valid handle is returned, the process is definitely running
+            // First try to OpenProcess by id, if valid handle is returned verify that process is running
             // Otherwise enumerate all processes and compare ids
```

Comment:
@adamsitnik, do you know why we fall back to the IndexOf here for the !IsRemoteMachine case?  I'm wondering why the using block below can't just contain `return !processHandle.IsInvalid && !HasExited(processHandle, ref signaled, out _);`.

Reply:
https://github.com/dotnet/runtime/pull/65041